dm/pkg/shardddl/optimism/operation.go

```diff
@@ -159,8 +168,14 @@ func PutOperation(cli *clientv3.Client, skipDone bool, op Operation, infoModRev
 // GetAllOperations gets all shard DDL operation in etcd currently.
 // This function should often be called by DM-master.
 // k/k/k/k/v: task-name -> source-ID -> upstream-schema-name -> upstream-table-name -> shard DDL operation.
-func GetAllOperations(cli *clientv3.Client) (map[string]map[string]map[string]map[string]Operation, int64, error) {
-	respTxn, _, err := etcdutil.DoOpsInOneTxnWithRetry(cli, clientv3.OpGet(common.ShardDDLOptimismOperationKeyAdapter.Path(), clientv3.WithPrefix()))
+func GetAllOperations(cli *clientv3.Client, task string) (map[string]map[string]map[string]map[string]Operation, int64, error) {
+	var queryKey string
+	if task == "" {
+		queryKey = common.ShardDDLOptimismOperationKeyAdapter.Path()
+	} else {
+		queryKey = common.ShardDDLOptimismOperationKeyAdapter.Encode(task)
```

Comment:
mixing Encode and WithPrefix will mix `task1` `task10` for current Encoder

Reply:
Can appending another "/" solve this problem?